observerevent event type issue fixed

RavindraChherke opened the pull request:
Looks like the MAKE_SYSTEM_PROP was not setting event values properly. I think keeping it in Module file is a good idea. Also, DATA_EVENT_TYPE_VALUE was set to FIRDataEventTypeChildAdded. Now changed to FIRDataEventTypeValue.

RavindraChherke committed:
observerevent event type issue fixed

Looks like the MAKE_SYSTEM_PROP was not setting event values properly. I think keeping it in Module file is a good idea. Also, DATA_EVENT_TYPE_VALUE was set to FIRDataEventTypeChildAdded. Now changed to FIRDataEventTypeValue.

hansemannn commented on ios/Classes/FirebaseDatabaseModule.m:
```diff
@@ -16,6 +16,12 @@ @implementation FirebaseDatabaseModule
+MAKE_SYSTEM_PROP(DATA_EVENT_TYPE_ADD, FIRDataEventTypeChildAdded);
+MAKE_SYSTEM_PROP(DATA_EVENT_TYPE_REMOVE, FIRDataEventTypeChildRemoved);
+MAKE_SYSTEM_PROP(DATA_EVENT_TYPE_MOVE, FIRDataEventTypeChildMoved);
+MAKE_SYSTEM_PROP(DATA_EVENT_TYPE_CHANGE, FIRDataEventTypeChildChanged);
```
Looks good! One typo I noticed (that was in my initial implementation as well) is that we should include the `CHILD` phrase as well, e.g.:
- `DATA_EVENT_TYPE_CHILD_ADDED`
- `DATA_EVENT_TYPE_CHILD_REMOVED`
- `DATA_EVENT_TYPE_CHILD_MOVED`
- `DATA_EVENT_TYPE_CHILD_CHANGED`

Can you change that in code and docs?

hansemannn reviewed (changes requested):
Left one review comment. Also, please bump the version in ios/manifest to the next patch version.

RavindraChherke committed:
Version number updated to 1.2

Observerevent name changed. Documentation updated.

RavindraChherke commented:
I have made the changes as suggested. Please check. Thanks!

hansemannn commented:
It should be a patch bump (1.1.1) instead of a minor bump (1.2.0), referring to the [semver specs](https://semver.org/). Also, the constants should be suffixed with `_CHANGED` instead of `_CHANGE`. Please update that last piece and attach the new module version. Thx!

RavindraChherke committed:
Version number correction 1.1.1

1. Typo fixed.
2. According to https://semver.org the patch version should be 1.1.1 Now it has been corrected.

RavindraChherke commented:
Sorry, I should pay more attention to details. This is my first time contributing to other project.

I have build the module with Titanium SDK 7.1.1  There are changes in titanium.xcconfig. Is that OK?

Please find the attached zip file. 

[firebase.database-iphone-1.1.1.zip](https://github.com/hansemannn/titanium-firebase-database/files/2034133/firebase.database-iphone-1.1.1.zip)

hansemannn committed:
Update titanium.xcconfig

hansemannn commented:
Looks perfect! PR **approved**. Keep up the great contributions to the community.

hansemannn commented:
New Release: https://github.com/hansemannn/titanium-firebase-database/releases/tag/ios-1.1.1
